lasy/profiles/spacetime_profile.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+
+from .profile import Profile
+
+
+class SpaceTimeProfile(Profile):
+    r"""
+    Class that can evaluate a pulse that has certain space-time couplings.
+
+    More precisely, the electric field corresponds to:
+
+    .. math::
+
+        E_u(\\boldsymbol{x}_\\perp,t) = Re\\left[ E_0\\,
+        \\exp\\left(-\\frac{\\boldsymbol{x}_\\perp^2}{w_0^2}
+        - \\frac{(t-t_{peak}+2ibx/w_0^2)^2}{\\tau_{eff}^2}
+        - i\\omega_0(t-t_{peak}) + i\\phi_{cep}\\right) \\times p_u \\right]
+
+    where :math:`u` is either :math:`x` or :math:`y`, :math:`p_u` is
+    the polarization vector, :math:`Re` represent the real part.
+    The other parameters in this formula are defined below.
+
+    Parameters
+    ----------
+    wavelength: float (in meter)
+        The main laser wavelength :math:`\\lambda_0` of the laser, which
+        defines :math:`\\omega_0` in the above formula, according to
+        :math:`\\omega_0 = 2\\pi c/\\lambda_0`.
+
+    pol: list of 2 complex numbers (dimensionless)
+        Polarization vector. It corresponds to :math:`p_u` in the above
+        formula ; :math:`p_x` is the first element of the list and
+        :math:`p_y` is the second element of the list. Using complex
+        numbers enables elliptical polarizations.
+
+    laser_energy: float (in Joule)
+        The total energy of the laser pulse. The amplitude of the laser
+        field (:math:`E_0` in the above formula) is automatically
+        calculated so that the pulse has the prescribed energy.
+
+    tau: float (in second)
+        The duration of the laser pulse, i.e. :math:`\\tau` in the above
+        formula. Note that :math:`\\tau = \\tau_{FWHM}/\\sqrt{2\\log(2)}`,
+        where :math:`\\tau_{FWHM}` is the Full-Width-Half-Maximum duration
+        of the intensity distribution of the pulse.
+
+    w0: float (in meter)
+        The waist of the laser pulse, i.e. :math:`w_0` in the above formula.
+
+    sc: spatial chirp, b in the above formula, that results in a mixing
+        of the longitudinal and transverse profiles. Must be in units
+        of [x/omega]. A representative real value is b = w0 * tau.

```

Comment:
Here is a suggestion that is more consistent with the other docstrings:
```suggestion
    b: float (in meter.second)
        Spatial chirp, i.e. :math:`b` in the above formula, that results in the laser frequency
        varying as a function of `x`. A representative real value is b = w0 * tau.

```